planner/core/find_best_task.go

```diff
@@ -430,6 +435,12 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty) (t task, err
 			}, nil
 		}
 		if path.isTablePath {
+			if ds.preferStoreType&preferTiFlash != 0 && path.storeType == kv.TiKV {
+				continue
+			}
+			if ds.preferStoreType&preferTiKV != 0 && path.storeType == kv.TiFlash {
+				continue
+			}
```

Comment:
We may end up with no plan generated if the hint specifies TiFlash while there is no TiFlash replica available?

Reply:
For now, if there is no TiFlash replica available, we still force a TiFlash plan.